File: Historical PowerTrack/Python/AcceptRejectHistoricalJob.py
# !/usr/bin/env python

import urllib2
import base64
import json


def post():

    url = 'ENTER_HISTORICAL_JOB_URL'
    username = 'ENTER_USERNAME'
    password = 'ENTER_PASSWORD'

    # Switch to 'reject' to reject the job.
    choice = 'accept'

    payload = {
        "status": choice
        }

    base64string = base64.encodestring("{0}:{1}".format(username, password))
    req = urllib2.Request(url=url, data=json.dumps(payload))
    req.add_header('Content-type', 'application/json')
    req.add_header("Authorization", "Basic {}".format(base64string.strip()))
    req.get_method = lambda: 'PUT'

    try:
        response = urllib2.urlopen(req)
        print(response.read())
    except urllib2.HTTPError as e:
        print(e)

    if __name__ == "__main__":
        post()

File: Historical PowerTrack/C#/AcceptRejectHistoricalJob.cs
using System;
using System.Net;
using System.IO;
using System.Text;

namespace BasicOps
{
	class MainClass
	{
		public static void Main (string[] args)
		{
			MainClass op = new MainClass();
			op.CreateHistoricalJob();
		}

		public HttpWebRequest makeRequest()
		{

			string urlString = "ENTER_JOB_URL_HERE"; // Enter the jobURL field for the historical job here.
			string username = "ENTER_USERNAME_HERE";
			string password = "ENTER_PASSWORD_HERE";

			HttpWebRequest request = (HttpWebRequest)WebRequest.Create(urlString);
            		request.ServicePoint.Expect100Continue = false;

		    	string authInfo = string.Format("{0}:{1}", username, password);
                        authInfo = Convert.ToBase64String(Encoding.Default.GetBytes(authInfo));
                        request.Headers.Add("Authorization", "Basic " + authInfo);
                
                        //In some Windows environments, this alternative Basic Authentication method is available.
                        //NetworkCredential nc = new NetworkCredential(username, password);
                        //request.Credentials = nc;
                        //request.PreAuthenticate = true;

			request.AutomaticDecompression = DecompressionMethods.Deflate | DecompressionMethods.GZip;

			return request;			
		}

		public void CreateHistoricalJob()
		{
			HttpWebRequest request = makeRequest();
			
			string postData = "";
            string choice = "accept"; // Switch to 'reject' to reject the job.

			request.Method = "PUT";
			postData = "{\"status\":\"" + choice + "\"}";

			byte[] byteArray = Encoding.UTF8.GetBytes (postData);
            		request.ContentType = "application/x-www-form-urlencoded";
	    		request.ContentLength = byteArray.Length;
            		Stream dataStream = request.GetRequestStream ();			
            		dataStream.Write (byteArray, 0, byteArray.Length);
            		dataStream.Close ();

            		WebResponse response = request.GetResponse ();
            		Console.WriteLine (((HttpWebResponse)response).StatusDescription);
            		dataStream = response.GetResponseStream ();
            		StreamReader reader = new StreamReader (dataStream);
            		string responseFromServer = reader.ReadToEnd ();
            		Console.WriteLine (responseFromServer);
				Console.WriteLine();
            		reader.Close ();
            		dataStream.Close ();
            		response.Close ();
		}		
	}
}
